packages/components/src/autocomplete/index.tsx

```diff
@@ -238,9 +237,10 @@ export function useAutocomplete( {
 	// but this is a preemptive performance improvement, since the autocompleter
 	// is a potential bottleneck for the editor type metric.
 	const textContent = useMemo( () => {
-		if ( isCollapsed( record ) ) {
+		if ( isCollapsed( record as RichTextValue ) ) {
```

Comment:
In this file, `record` is a type which makes sure that a couple properties of `RichTextValue` can't be null. As far as I can tell, that type is fully compatible with `RichTextValue`, so we can just cast to it on the boundary of components and rich text here. (And elsewhere in the file.)

Reply:
I personally find that the need for `as` can often be an indication that something could be wrong in our code, or that at least it could be rewritten in a way that doesn't required type casting.

On a similar note to [@tyxla 's comment](https://github.com/WordPress/gutenberg/pull/49651/files#r1160670874), could we reuse types from the rich text package for the [`record` prop](https://github.com/WordPress/gutenberg/blob/bdd50b5a8a86dedcbc81cde4807a8fe47115cad7/packages/components/src/autocomplete/types.ts) ?